src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java

```diff
@@ -270,30 +306,83 @@ public Page<FeederResultEntity> findFeederResultsPage(ShortCircuitAnalysisResult
         // cf. https://vladmihalcea.com/fix-hibernate-hhh000104-entity-fetch-pagination-warning-message/
         // We must separate in two requests, one with pagination the other one with Join Fetch
-        Page<FaultResultEntity> faultResultsPage = faultResultRepository.findAll(specification, addDefaultSort(pageable, DEFAULT_FAULT_RESULT_SORT_COLUMN));
+        Page<FaultResultEntity> faultResultsPage = faultResultRepository.findAll(
+                specification,
+                addDefaultSort(
+                        filterOutChildrenSort(pageable, secondarySort),
+                        DEFAULT_FAULT_RESULT_SORT_COLUMN
+                ));
         if (faultResultsPage.hasContent()) {
-            appendLimitViolationsAndFeederResults(faultResultsPage);
+            appendLimitViolationsAndFeederResults(faultResultsPage, secondarySort, resourceFilters);
         }
         return faultResultsPage;
     }
 
-    private void appendLimitViolationsAndFeederResults(Page<FaultResultEntity> pagedFaultResults) {
+    private void appendLimitViolationsAndFeederResults(Page<FaultResultEntity> pagedFaultResults,
+                                                       Sort.Order secondarySort,
+                                                       List<ResourceFilter> resourceFilters) {
         // using the Hibernate First-Level Cache or Persistence Context
         // cf.https://vladmihalcea.com/spring-data-jpa-multiplebagfetchexception/
         if (!pagedFaultResults.isEmpty()) {
             List<UUID> faultResultsUuids = pagedFaultResults.stream()
                     .map(FaultResultEntity::getFaultResultUuid)
-                    .collect(Collectors.toList());
+                    .toList();
+
             faultResultRepository.findAllWithLimitViolationsByFaultResultUuidIn(faultResultsUuids);
+
             faultResultRepository.findAllWithFeederResultsByFaultResultUuidIn(faultResultsUuids);
+            sortFeeders(pagedFaultResults, secondarySort);
+            filterFeeders(pagedFaultResults, resourceFilters);
+        }
+    }
+
+    private void filterFeeders(Page<FaultResultEntity> pagedFaultResults, List<ResourceFilter> resourceFilters) {
+        // feeders may only be filtered through connectableId
+        Optional<ResourceFilter> connectableIdFilter = resourceFilters.stream()
+                .filter(filter -> FeederResultEntity.Fields.connectableId.equals(filter.column()))
+                .findFirst();
+        connectableIdFilter.ifPresent(resourceFilter -> pagedFaultResults
+                .map(faultRes -> {
+                    List<FeederResultEntity> feeders = faultRes.getFeederResults()
+                            .stream()
+                            .filter(feeder -> feeder.match(resourceFilter)
+                            ).toList();
+                    faultRes.setFeederResults(feeders);
+                    return faultRes;
+                }));
+    }
```

Comment:
filtering children can be made in JPA

Reply:
Indeed and thanks a lot for the help. Done.